landscaper/common/generate/openapi/generate.go

```diff
@@ -0,0 +1,307 @@
+	}
+
+	// execute the binary and log its outputs
+	return executeOpenAPIGen(binaryPath, outputDirectory, licensePath, verbosity, inputPackages)
+}
+
+// executeOpenAPIGen executes the `openapi-gen` binary to generate OpenAPI go-code
+func executeOpenAPIGen(executablePath string, outputDirectory string, licensePath string, verbosity string, inputPackages []string) error {
+	outputPackageName := "openapi"
+	reportFilename := fmt.Sprintf("%s/%s/api_violations.report", outputDirectory, outputPackageName)
+
+	// default license path
+	if len(licensePath) == 0 {
+		licensePath = fmt.Sprintf("%s/hack/LICENSE_BOILERPLATE.txt", rootDirectory)
+	}
+
+	args := []string{
+		"--v",
+		verbosity,
+		"--logtostderr",
+		fmt.Sprintf("--report-filename=%s", reportFilename),
+		fmt.Sprintf("--output-package=%s", outputPackageName),
+		fmt.Sprintf("--output-base=%s", outputDirectory),
+		fmt.Sprintf("-h"),
+		licensePath,
+	}
+
+	inputDirs := []string{}
+	for _, in := range inputPackages {
+		inputDirs = append(inputDirs, fmt.Sprintf("--input-dirs=%s", in))
+	}
+	args = append(args, inputDirs...)
+
+	cmd := exec.Command(executablePath, args...)
+
+	// the binary logs to STDERR
+	stderr, err := cmd.StderrPipe()
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Starting OpenAPI generation to directory: %s \n", outputDirectory)
+	err = cmd.Start()
+	if err != nil {
+		return err
+	}
+
+	// print the output of the subprocess
+	scanner := bufio.NewScanner(stderr)
+	for scanner.Scan() {
+		m := scanner.Text()
+		fmt.Printf("%s\n", m)
+	}
+
+	if err := cmd.Wait(); err != nil {
+		return fmt.Errorf("error waiting for openapi-gen: %+v", err)
+	}
+
+	fmt.Printf("OpenAPI generation successful. \n")
+
+	return nil
+}
+
+// parseImportPackages recursively parses the import packages of a given inputDirectory
+// returns the union of all found packages or an error
+func parseImportPackages(inputDirectory string, rootDirectory string, alreadyParsedPackages sets.String, filterPackages sets.String) ([]string, error) {
+	fset := token.NewFileSet()
+	foo, err := parser.ParseDir(fset, inputDirectory, func(info os.FileInfo) bool {
+		if info.IsDir() {
+			return false
+		}
+
+		if filteredFileNames.Has(info.Name()) || strings.HasSuffix(info.Name(), "_test.go") || strings.HasPrefix(info.Name(), "zz_") {
+			return false
+		}
+
+		return true
+	}, parser.ImportsOnly)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(foo) == 0 {
+		return nil, nil
+	}
+
+	if len(foo) > 1 {
+		return nil, fmt.Errorf("found multiple packages in input directory: %s", inputDirectory)
+	}
+
+	inputPackages := sets.NewString()
+	for _, v := range foo {
+		for _, valueFile := range v.Files {
+			for _, i := range valueFile.Imports {
+				escapedPackage := strings.ReplaceAll(i.Path.Value, "\"", "")
+				isK8sPackage := strings.HasPrefix(escapedPackage, "k8s.io")
+				isGithubPackage := strings.HasPrefix(escapedPackage, "github.com")
+
+				if !isGithubPackage && !isK8sPackage {
```

Comment:
Why are you skipping other packages?

Reply:
Mainly to speed up the generation process. This openapi annotation on the golang types are kube-openapi specific, hence Kubernetes specific.
If we find packages in the future that contain K8s types that are not either under `k8s.io` or `github.com` we can easily add a flag/argument to the program later. 
Does that sound ok?

Reply:
perfect :-)